presto-orc/src/main/java/com/facebook/presto/orc/cache/StorageOrcFileTailSource.java

```diff
@@ -40,7 +40,8 @@
 {
     private static final Logger log = Logger.get(StorageOrcFileTailSource.class);
 
-    private static final int EXPECTED_FOOTER_SIZE = 16 * 1024;
+    // try to catch both footer and metadata cache with a large enough first read
+    private static final int EXPECTED_FOOTER_SIZE = 1024 * 1024;
```

Comment:
Increased this value to increase the chance of reading metadata + footer + postscript in one IO call. We use the same value in bbio.

Reply:
In general, I would be worried about the memory consumption of the last commit. Shall we have some workload testing + go through each memory bump to either count it or make it configurable?